src/core/thread/mle_router.cpp

```diff
@@ -1309,7 +1310,7 @@ ThreadError MleRouter::HandleAdvertisement(const Message &aMessage, const Ip6::M
     case kDeviceStateChild:
         if ((mDeviceMode & ModeTlv::kModeFFD) && (GetActiveRouterCount() < mRouterUpgradeThreshold))
         {
-            BecomeRouter(ThreadStatusTlv::kTooFewRouters);
+            mRouterSelectionJitterTimer.Start(Timer::SecToMsec(otPlatRandomGet() % kRouterSelectionJitter));
```

Comment:
5.9.8 Processing Route64 TLVs (page114: line23 to 25) first mentions this, then 5.9.9 Router ID mgmt section (page:115 line 5 to 10) describes in details. And considering from the purpose of ROUTER_SELECTION_JITTER, it's used to prevent multiple children from requesting Router ID at the same time. For child Id Response, it's a unicast message.